domain/consensus/processes/pruningmanager/pruningmanager.go

```diff
@@ -310,16 +311,20 @@ func (pm *pruningManager) savePruningPoint(pruningPointHash *externalapi.DomainH
 	onEnd := logger.LogAndMeasureExecutionTime(log, "pruningManager.savePruningPoint")
 	defer onEnd()
 
-	utxoIter, err := pm.consensusStateManager.RestorePastUTXOSetIterator(pruningPointHash)
+	utxoSetIterator, err := pm.consensusStateManager.RestorePastUTXOSetIterator(pruningPointHash)
```

Comment:
We should make sure `RestorePastUTXOSetIterator` doesn't keep too much memory in RAM here

Reply:
Strictly speaking, I didn't touch the non-IBD parts of pruning in this ticket.
You're right though this is would indeed eat up O(UTXOSet). Maybe we want to open a ticket about it